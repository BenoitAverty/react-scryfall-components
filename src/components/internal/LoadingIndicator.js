import React from 'react';
import styled from 'react-emotion';

const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  & > div {
    display: inline-block;
    position: relative;
    width: 64px;
    height: 64px;

    & > div {
      display: inline-block;
      position: absolute;
      left: 6px;
      width: 13px;
      background: #fff;
      animation: lds-facebook 1.2s cubic-bezier(0, 0.5, 0.5, 1) infinite;
    }
    & > div:nth-child(1) {
      left: 6px;
      animation-delay: -0.24s;
    }
    & > div:nth-child(2) {
      left: 26px;
      animation-delay: -0.12s;
    }
    & > div:nth-child(3) {
      left: 45px;
      animation-delay: 0;
    }
    @keyframes lds-facebook {
      0% {
        top: 6px;
        height: 51px;
      }
      50%,
      100% {
        top: 19px;
        height: 26px;
      }
    }
  }
`;

const LoadingIndicator = () => (
  <Container>
    <div>
      <div />
      <div />
      <div />
    </div>
  </Container>
);

export default LoadingIndicator;
