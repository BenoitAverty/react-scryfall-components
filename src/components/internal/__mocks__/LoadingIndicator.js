import React from 'react';

export default () => <div data-testid="LoadingIndicatorStub" />;
