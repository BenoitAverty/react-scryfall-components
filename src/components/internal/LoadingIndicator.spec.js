import React from 'react';
import { render } from '@testing-library/react';

import LoadingIndicator from './LoadingIndicator';

test('basic snapshot', () => {
  const { container } = render(<LoadingIndicator />);

  expect(container.firstChild).toMatchSnapshot();
});

test('width / height snapshot', () => {
  const { container } = render(<LoadingIndicator width={120} height={40} />);

  expect(container.firstChild).toMatchSnapshot();
});
