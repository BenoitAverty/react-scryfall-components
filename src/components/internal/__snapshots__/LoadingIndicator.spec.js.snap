// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<div
  class="css-80iyfg e1i2ck7j0"
  height="100%"
  width="100%"
>
  <div>
    <div />
    <div />
    <div />
  </div>
</div>
`;

exports[`width / height snapshot 1`] = `
<div
  class="css-126bhm e1i2ck7j0"
  height="40px"
  width="120px"
>
  <div>
    <div />
    <div />
    <div />
  </div>
</div>
`;
